app/src/main/java/org/jellyfin/androidtv/model/trailers/lifter/YouTubeExternalTrailerLifter.kt

```diff
@@ -1,21 +1,58 @@
 package org.jellyfin.androidtv.model.trailers.lifter
 
-import android.net.Uri
+import android.util.Log
 import org.jellyfin.androidtv.model.trailers.external.YouTubeTrailer
 import org.jellyfin.apiclient.model.entities.MediaUrl
+import java.net.URL
 
 class YouTubeExternalTrailerLifter : ExternalTrailerLifter() {
+	private val LOG_TAG = "YouTubeLifter"
+
 	private val youtubeDomains = arrayListOf("youtube.com", "youtu.be")
 
-	private fun isYoutubeUrl(uri: Uri) = youtubeDomains.contains(getDomain(uri))
-	private fun isYoutubeUrl(url: MediaUrl) = isYoutubeUrl(Uri.parse(url.url))
+	private fun isYoutubeUrl(url: URL) = youtubeDomains.contains(getDomain(url))
+
+	private val videoParameterRegex = Regex("v=([a-zA-Z0-9]+)")
 
 	override fun canLift(url: MediaUrl): Boolean {
-		return isYoutubeUrl(url)
+		val converted = mediaUrlToUrl(url)
+		return converted != null && isYoutubeUrl(converted)
 	}
 
-	override fun lift(url: MediaUrl): YouTubeTrailer {
-		return YouTubeTrailer(url.name, url.url, "")
+	override fun lift(url: MediaUrl): YouTubeTrailer? {
+		if (!canLift(url)) {
+			 Log.e(LOG_TAG, "URL ${url.url} is not supported")
+			return null
+		}
+
+		val convertedURL = mediaUrlToUrl(url)!!
+
+		val videoKey = when (getDomain(convertedURL)!!) {
+			"youtube.com" -> {
+				val matchGroups = videoParameterRegex.find(convertedURL.query)?.groups
+				if (matchGroups == null || matchGroups.size != 2) {
+					Log.e(LOG_TAG, "URL $convertedURL is not supported")
+					return null
+				} else {
+					matchGroups[1]!!.value
+				}
+			}
+
+			"youtu.be" -> {
+				convertedURL.path.subSequence(1, convertedURL.path.length)
```

Comment:
I'd say use the last 11 characters from the path in the case a url looks like `youtu.be/sample/[id]`